flocker/control/test/test_model_change.py

```diff
@@ -0,0 +1,418 @@
+        internally changed then the output changes.
+        """
+        class Original(CheckedPSet):
+            __type__ = Subtype
+
+        class Different(CheckedPSet):
+            __type__ = ChangedSubtype
+        Different.__name__ = "Original"
+
+        self.assert_catches_changes(Original, Different)
+
+    def test_pvector_value_new_type(self):
+        """
+        If the type of the value of a ``PVector`` changes the output changes.
+        """
+        class Original(CheckedPVector):
+            __type__ = int
+
+        class Different(CheckedPVector):
+            __type__ = str
+        Different.__name__ = "Original"
+
+        self.assert_catches_changes(Original, Different)
+
+    def test_pvector_value_type_changed(self):
+        """
+        If the type of the value of a ``PVector`` is the same, but it has
+        internally changed then the output changes.
+        """
+        class Original(CheckedPVector):
+            __type__ = Subtype
+
+        class Different(CheckedPVector):
+            __type__ = ChangedSubtype
+        Different.__name__ = "Original"
+
+        self.assert_catches_changes(Original, Different)
+
+
+def persist_model():
+    """
+    Store the current model to disk.
+
+    We also store the git hash of current checkout, so it's clear what
+    version of code was used to generate the model.
+    """
+    git_hash = check_output(
+        [b"git", b"show", b'--format=%H', b"-s"]).strip()
+    model = generate_model()
+    PERSISTED_MODEL.setContent(dumps(
+        {u"git_hash": git_hash, u"model": model}, sort_keys=True, indent=4,
+        separators=(',', ': ')))
+    print("Persisted model for git hash {}.".format(git_hash))
+
+
+class ConfigurationModelChanged(SynchronousTestCase):
+    """
+    Detect when the configuration model has changed.
+    """
+    def test_model_changed(self):
+        """
+        Most of the time if the configuration model changes this test will
+        fail.
+
+        This does not indicate a bug. Rather, it indicates that you should
+        implement upgrade code for the on-disk configuration. Once you are
+        confident it is possible to upgrade from older versions of Flocker
+        to the new version of the code you have introduced, you can update
+        this test by running:
+
+            $ python -m flocker.control.test.test_model_change
+
+        And then committing the resulting changes to git.
+
+        Note that this test may *not* fail in some cases where you still
+        need to write upgrade code, so don't rely on it to always tell you
+        when you need to write upgrade code.
+        """
+        current_model = generate_model()
+        previous_model = loads(PERSISTED_MODEL.getContent())[u"model"]
+        self.assertEqual(current_model, previous_model,
+                         self.test_model_changed.__doc__)
+
+
+if __name__ == '__main__':
+    persist_model()
```

Comment:
The convention for running a test module is to run the tests in it.  Not sure this is a problem here.  The alternative, making a real script for this functionality, is certainly more work (and since we probably don't want to distribute/install the script alongside the other scripts ... I don't even really know how we would do it ...).  Just thought it was worth a comment.